core/src/main/java/com/linecorp/armeria/common/unsafe/PooledHttpData.java

```diff
@@ -38,24 +47,30 @@
  * you should only use this class in performance-sensitive situations and after being ready to deal with these
  * very hard-to-debug issues.
  *
- * <p>You may interact with {@link PooledHttpData} when using objects that return pooled objects, such as
- * {@link PooledWebClient}. If you don't use such objects, you will never see a {@link PooledHttpData} and don't
- * need to read further.
+ * <p>You may interact with {@link PooledHttpData} when using operations that return pooled objects, such as:
+ * <ul>
+ *   <li>{@link StreamMessage#subscribe(Subscriber, SubscriptionOption...)} with
+ *       {@link SubscriptionOption#WITH_POOLED_OBJECTS}</li>
+ *   <li>{@link HttpRequest#aggregateWithPooledObjects(ByteBufAllocator)}</li>
+ *   <li>{@link HttpResponse#aggregateWithPooledObjects(ByteBufAllocator)}</li>
+ *   <li>{@link HttpFile#aggregateWithPooledObjects(Executor, ByteBufAllocator)}</li>
+ * </ul>
+ * If you don't use such operations, you will never see a {@link PooledHttpData} and don't need to read further.
  *
  * <h3>Impact of pooled buffers</h3>
  *
- * <p>Any time you receive a {@link PooledHttpData} it will have a single reference that must be released -
- * failure to release the reference will result in a memory leak and poor performance. You must make sure to do
- * this by calling {@link PooledHttpData#close()}, usually in a try-with-resources structure to avoid side
- * effects.
+ * <p>Any time you receive a {@link PooledHttpData} it will have a single reference of {@link ByteBuf} that
+ * must be released - failure to release the {@link ByteBuf} will result in a memory leak and poor performance.
+ * You must make sure to do this by calling {@link PooledHttpData#close()}, usually in a try-with-resources
+ * structure to avoid side effects.
  *
  * <p>For example, <pre>{@code
- * PooledHttpResponse res = PooledHttpResponse.of(client.get("/"));
+ * HttpResponse res = client.get("/");
  * res.aggregateWithPooledObjects(ctx.alloc(), ctx.executor())
  *    .thenApply(aggResp -> {
  *        // try-with-resources here ensures the content is released
  *        // if it is a ByteBufHttpData, or otherwise is a no-op if it is not.
- *        try (PooledHttpData content = aggResp.content()) {
+ *        try (PooledHttpData content = (PooledHttpData) aggResp.content()) {
```

Comment:
I don't think a user can be guaranteed to have a `PooledHttpData` even when aggregating with pooled objects if downstream doesn't pool (wrapping into pooled was one of the features of the new API so users don't have to use `instanceof` for safety). Are you going to make `DefaultHttpData` implement `PooledHttpData` too as a no-op? Otherwise, I think we can remove `SafeCloseable` and potentially `PooledHttpData`, it doesn't add anything over `ByteBufHolder`

Reply:
Let me handle this in a follow-up PR I'm working on now.